site/app/controllers/admin/NotebookBuilderController.php

```diff
@@ -304,4 +331,89 @@ private function checkPermissions(string $autograding_dir): array {
 
         return $errors;
     }
+
+    /** Convert the JSON file in notebook render format
+     * @param string $filepath The path of the notebook JSON data.
+     */
+    private function renderNotebookPreview(string $filepath){
```

Comment:
This seems to be largely copied from the Jupyter Notebook conversion code I wrote.  Given that Submitty Notebooks support more than Jupyter Notebooks, this is not sufficient.  Instead, it would be better to use `Notebook::getNotebook()` or similar.

Reply:
I will remove this part.